core/src/main/java/org/apache/calcite/jdbc/CalciteSchema.java

```diff
@@ -322,7 +322,6 @@ public final NavigableMap<String, CalciteSchema> getSubSchemaMap() {
     final ImmutableSortedMap.Builder<String, CalciteSchema> builder =
         new ImmutableSortedMap.Builder<>(NameSet.COMPARATOR);
     builder.putAll(subSchemaMap.map());
-    addImplicitSubSchemaToBuilder(builder);
```

Comment:
@kramerul Sorry, I'm late to the party here. Quick question: isn't this a breaking change? Previously `getSubSchemaMap` would return both the explicit and implicit sub-schema (as its javadoc states), but now the implicit ones would no longer be considered? (I guess those could still be accessed via the new `subSchemas()` method), but it'd seem that `getSubSchemaMap` has changed its behavior (and its javadoc is no longer accurate). Or I'm missing something?

Reply:
Hi @rubenada,

you are completely right. This is a breaking change. 

`getSubSchemaMap` needs to be changed to read all sub schemas using `subSchemas`.

Reply:
I think it should look like this

```java
  public final NavigableMap<String, CalciteSchema> getSubSchemaMap() {
    // Build a map of implicit sub-schemas first, then explicit sub-schemas.
    // If there are implicit and explicit with the same name, explicit wins.
    final ImmutableSortedMap.Builder<String, CalciteSchema> builder =
        new ImmutableSortedMap.Builder<>(NameSet.COMPARATOR);
    Lookup<CalciteSchema> schemas = subSchemas();
    for ( String name : schemas.getNames(LikePattern.any())) {
      builder.put(name, schemas.get(name));
    }
    return builder.build();
  }
```

Reply:
Thanks for the confirmation @kramerul , I'll create a PR accordingly

Reply:
No problem @kramerul ! Thanks you for suggesting the code to fix it :)
If you have a bit of time, could you please take a look at https://github.com/apache/calcite/pull/4254 to confirm it's ok?